drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/DMNTypeServiceImpl.java

```diff
@@ -85,103 +91,179 @@ public DMNRuntime getDMNRuntime(Path path) {
+        if (!type.isCollection()) {
+            throw new WrongDMNTypeException();
         }
+        populateGeneric(genericTypeInfoMap, "value", type.getName());
+        FactModelTree toReturn = createFactModelTreeForSimple(genericTypeInfoMap, factName, List.class.getCanonicalName(), fmType);
+        FactModelTree genericFactModelTree = createFactModelTreeForGenericType(new HashMap<>(), type.getName(), type.getName(), type.getName(), type, hiddenFacts, fmType);
+        hiddenFacts.put(type.getName(), genericFactModelTree);
+        return toReturn;
+    }
 
-        Map<String, String> simpleFields = new HashMap<>();
-        Map<String, List<String>> genericTypeInfoMap = new HashMap<>();
-        FactModelTree factModelTree = new FactModelTree(name, "", simpleFields, genericTypeInfoMap, fmType);
-        for (Map.Entry<String, DMNType> entry : type.getFields().entrySet()) {
-
-            String expandableId = path + "." + entry.getKey();
-
-            // if it is a collection, just generate the generic and add as hidden fact a simple or composite fact model tree
-            if (entry.getValue().isCollection()) {
-                String genericKey = populateGeneric(simpleFields, genericTypeInfoMap, path, entry.getValue().getName(), entry.getKey());
-
-                FactModelTree fact = createFactModelTree(entry.getKey(), expandableId, entry.getValue(), hiddenFacts, FactModelTree.Type.UNDEFINED, true);
-                hiddenFacts.put(genericKey, fact);
-            }
-            // a simple type is just name -> type
-            else if (!entry.getValue().isComposite()) {
-                simpleFields.put(entry.getKey(), entry.getValue().getName());
-            }
-            // a complex type needs the expandable property and then in the hidden map, its fact model tree
-            else {
-                factModelTree.addExpandableProperty(entry.getKey(), expandableId);
-                hiddenFacts.put(expandableId, createFactModelTree(entry.getKey(), expandableId, entry.getValue(), hiddenFacts, FactModelTree.Type.UNDEFINED));
+    /**
+     * Creates a <code>FactModelTree</code> for <code>DMNType</code> where <code>DMNType.isCollection()</code> != <code>true</code>
+     * @param propertyName
+     * @param fullPropertyPath
+     * @param type
+     * @param hiddenFacts
+     * @param fmType
+     * @return
+     * @throws WrongDMNTypeException if <code>DMNType.isCollection()</code> == <code>true</code>
+     */
+    protected FactModelTree createFactModelTreeForNoCollection(Map<String, List<String>> genericTypeInfoMap, String factName, String propertyName, String fullPropertyPath, DMNType type, SortedMap<String, FactModelTree> hiddenFacts, FactModelTree.Type fmType) throws WrongDMNTypeException {
+        if (type.isCollection()) {
+            throw new WrongDMNTypeException();
+        }
+        boolean isComposite = type.isComposite();
+        if (isComposite) {
```

Comment:
Please add a comment like 
// Context is a special case that is considered as composite but has no nested fields so it should be considered as simple for editing

Reply:
Done